js/router.js

```diff
@@ -1,15 +1,147 @@
+import $ from 'jquery';
 import { Router } from 'backbone';
+import { getGuid } from './utils';
+import { getPageContainer } from './utils/selectors';
+import UserPage from './views/UserPage';
+import TransactionsPage from './views/TransactionsPage';
+import TemplateOnly from './views/TemplateOnly';
 
 export default class ObRouter extends Router {
-  constructor() {
+  constructor(options = {}) {
+    super(options);
+    this.options = options;
+
+    // Temporarily requiring a dummy 'users' collection. Eventually, we'll
+    // be getting this data via fetches to the server
+    if (!options.usersCl) {
+      throw new Error('Need me some dummy users!');
+    }
+
+    this.usersCl = options.usersCl;
+
+    if (!options.pageNavVw) {
+      throw new Error('Please provide the app\'s pageNavVw.');
+    }
+
+    this.pageNavVw = options.pageNavVw;
+
     const routes = [
-      // ['home/:state(/:searchText)', 'home'],
-      // ['userPage/:userID(/:state)(/:itemHash)(/:skipNSFWmodal)', 'userPage'],
-      // [/^@([^\/]+)(.*)$/, 'userPageViaHandle'],
-      // ['transactions/:state(/:orderID)(/:tabState)', 'transactions'],
-      // ['settings/:state', 'settings'],
+      [/^@([^\/]+)[\/]?([^\/]*)[\/]?([^\/]*)[\/]?([^\/]*)$/, 'userViaHandle'],
+      [/^(Qm[a-zA-Z0-9]+)[\/]?([^\/]*)[\/]?([^\/]*)[\/]?([^\/]*)$/, 'user'],
+      ['transactions', 'transactions'],
+      ['transactions/:tab', 'transactions'],
+      ['*path', 'pageNotFound'],
     ];
 
-    routes.forEach((route) => this.route.apply(this, route));
+    routes.slice(0)
+      .reverse()
+      .forEach((route) => this.route.apply(this, route));
+
+    this.setAddressBarText();
+
+    $(window).on('hashchange', () => {
```

Comment:
The router is updating the address bar in the nav bar based upon the window's hashchanges. So, views shouldn't have to directly update that. All they need to do is to use the routers `navigate` method when their state should be represented by a different url, for example:

```
app.router.navigate('@themes/store')
```